extensions/amp-brightcove/1.0/component.js

```diff
@@ -0,0 +1,225 @@
+    () =>
+      addParamsToUrl(
+        `https://players.brightcove.net/${encodeURIComponent(account)}` +
+          `/${encodeURIComponent(player)}` +
+          `_${encodeURIComponent(embed)}/index.html` +
+          '?amp=1' +
+          // These are encodeURIComponent'd in encodeId_().
+          (playlistId
+            ? '&playlistId=' + encodeId(playlistId)
+            : videoId
+            ? '&videoId=' + encodeId(videoId)
+            : ''),
+        {...urlParams, referrer, playsinline: true, autoplay: undefined}
+      ),
+    [account, embed, player, playlistId, referrer, urlParams, videoId]
+  );
+
+  const makeMethodMessage = useCallback((method) => makeMessage(method), []);
+  const onMessage = useCallback(
+    ({currentTarget, data}) => {
+      const eventType = data?.event;
+      switch (eventType) {
+        case 'ready':
+          onReady?.(data, player);
+          break;
+        case 'playing':
+          onPlayingState?.(true);
+          break;
+        case 'pause':
+        case 'ended':
+          onPlayingState?.(false);
+          break;
+      }
+
+      if (data?.['ct']) {
+        playerStateRef.current['currentTime'] = data['ct'];
+      }
+      if (data?.['pr']) {
+        playerStateRef.current['playedRanges'] = data['pr'];
+      }
+      if (data?.['dur']) {
+        playerStateRef.current['duration'] = data['dur'];
+      }
+
+      if (events[eventType]) {
+        dispatchCustomEvent(currentTarget, events[eventType]);
+        return;
+      }
+
+      if (eventType === 'volumechange') {
+        const isMuted = data['muted'];
+        if (isMuted == null || isMuted == muted) {
+          return;
+        }
+        setMuted(isMuted);
+        dispatchCustomEvent(currentTarget, mutedOrUnmutedEvent(isMuted));
+      }
+    },
+    [muted, onReady, player, onPlayingState]
+  );
+
+  // Check for valid props
+  if (!checkProps(account)) {
+    return null;
+  }
+
+  return (
+    <VideoIframe
+      ref={ref}
+      {...rest}
+      autoplay={autoplay}
+      makeMethodMessage={makeMethodMessage}
+      muted={muted}
+      onIframeLoad={(event) => {
+        const {currentTarget} = event;
+        dispatchCustomEvent(currentTarget, 'canplay');
+      }}
+      onMessage={onMessage}
+      playerStateRef={playerStateRef}
+      src={src}
+    />
+  );
+}
+const Brightcove = forwardRef(BrightcoveWithRef);
+Brightcove.displayName = 'Brightcove'; // Make findable for tests.
+export {Brightcove};
+
+/**
+ * id is either a Brightcove-assigned id, or a customer-generated
+ * reference id. Reference ids are prefixed 'ref:' and the colon
+ * must be preserved unencoded.
+ * @param {string} id
+ * @return {string}
+ */
+function encodeId(id) {
+  /*  */
+  if (id.substring(0, 4) === 'ref:') {
+    return `ref:${encodeURIComponent(id.substring(4))}`;
+  }
+  return encodeURIComponent(id);
+}
+
+/**
+ * @param {string} method
+ * @param {*=} arg
+ * @return {string}
+ */
+function makeMessage(method, arg) {
```

Comment:
what is `arg` for? its explicitly ignored in the `useCallback` above? should it be a local var instead of a parameter?

Reply:
Agreed with local var.

Reply:
Done, thanks